File: share/pegasus/init/daxgen/DAXGen.java
import edu.isi.pegasus.planner.dax.*;

/* API Documentation: http://pegasus.isi.edu/documentation */
public class DAXGen {

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: DAXGen DAXFILE");
            System.exit(1);
        }

        String daxfile = args[0];

        ADAG dax = new ADAG("{{name}}");

        // TODO Add some jobs
        /*
        Job j = new Job("ID1", "myexe");
        File a = new File("a.txt");
        File b = new File("b.txt");
        File c = new File("c.txt");
        j.addArgument("-i").addArgument(a);
        j.addArgument("-o").addArgument(b);
        j.addArgument("-o").addArgument(c);
        j.uses(a, File.LINK.INPUT, File.TRANSFER.TRUE, false);
        j.uses(b, File.LINK.OUTPUT, File.TRANSFER.TRUE, false);
        j.uses(c, File.LINK.OUTPUT, File.TRANSFER.TRUE, false);
        dax.addJob(j);
        */

        // TODO Add dependencies
        /*
        dax.addDependency(j, k);
        */

        dax.writeToFile(daxfile);
    }
}

File: share/pegasus/init/daxgen/daxgen.pl
use 5.006;
use warnings;
use strict;
use Pegasus::DAX::Factory qw(:all);

# API Documentation: http://pegasus.isi.edu/documentation

scalar(@ARGV) == 1 or die "Usage: daxgen.py DAXFILE\n";

my $daxfile = $ARGV[0];

my $dax = newADAG(name => '{{name}}');

# TODO Add some jobs
#my $j = newJob(name => 'myexe');
#my $a = newFilename(name => 'a.txt', (link => LINK_IN, transfer => 'true', register => 'false'));
#my $b = newFilename(name => 'b.txt', (link => LINK_OUT, transfer => 'true', register => 'false'));
#my $c = newFilename(name => 'c.txt', (link => LINK_OUT, transfer => 'true', register => 'false'));
#$j->addArgument('-i', $a, '-o', $b, '-o', $c);
#$dax->addJob($j);

# TODO Add some dependencies
#$dax->addDependency($j, $k);

open my $f, '>', $daxfile or die "Unable to open DAXFILE: $!\n";
$dax->toXML($f);
close($f);

